stx-rosetta.Dockerfile

```diff
@@ -1,147 +1,124 @@
 ENV STACKS_BLOCKCHAIN_API_HOST=0.0.0.0
-
 ENV STACKS_CORE_RPC_HOST=127.0.0.1
 ENV STACKS_CORE_RPC_PORT=20443
-
-### Startup script & coordinator
-RUN printf '#!/bin/bash\n\
-trap "exit" INT TERM\n\
-trap "kill 0" EXIT\n\
-echo Your container args are: "$@"\n\
-tail --retry -F stacks-api.log stacks-node.log 2>&1 &\n\
-while true\n\
-do\n\
-  pg_start\n\
-  stacks_api &> stacks-api.log &\n\
-  stacks_api_pid=$!\n\
-  if [ $STACKS_NETWORK = "mocknet" -o $STACKS_NETWORK = "dev" ]; then\n\
-    stacks-node start --config=/data/stacky/Stacks-${STACKS_NETWORK}.toml &> stacks-node.log &\n\
-  elif [ $STACKS_NETWORK = "testnet"]; then \n\
-    stacks-node start --config=/data/stacky/Stacks-mocknet.toml &> stacks-node.log &\n\
-  else\n\
-    stacks-node mainnet &> stacks-node.log &\n\
-  fi\n\
-  stacks_node_pid=$!\n\
-  wait $stacks_node_pid\n\
-  echo "node exit, restarting..."\n\
-  rkill -9 $stacks_api_pid\n\
-  pg_stop\n\
-  rm -rf $PGDATA\n\
-  sleep 5\n\
-done\n\
-' >> run.sh && chmod +x run.sh
-
+ENV MAINNET_STACKS_CHAIN_ID=0x00000001
+ENV TESTNET_STACKS_CHAIN_ID=0x80000000
+ENV V2_POX_MIN_AMOUNT_USTX=90000000260
+RUN apt-get update \
+    && apt install -y \
+        gnupg2 \
+        lsb-release \
+        curl procps \
+        netcat \
+        gosu \
+        runit-init \
+        rsyslog
+RUN curl -sL https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - \
+    && echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" > /etc/apt/sources.list.d/pgsql.list \
+    && curl -sL https://deb.nodesource.com/setup_14.x | bash -
+RUN apt-get update \
+    && apt-get install -y \
+        postgresql-${PG_VERSION} \
+        postgresql-client-${PG_VERSION} \
+        nodejs \
+    && echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+RUN mkdir -p \
+        ${STACKS_SVC_DIR}/postgresql/log \
+        ${STACKS_SVC_DIR}/stacks-blockchain-api/log \
+        ${STACKS_SVC_DIR}/stacks-blockchain \
+        ${STACKS_LOG_DIR}/postgresql \
+        ${STACKS_LOG_DIR}/stacks-blockchain-api/log \
+    && apt-get clean \
+    && rm -rf /var/cache/apt/* /var/lib/apt/lists/* /tmp/* ${STACKS_SVC_DIR}/getty*
+COPY --from=stacks-blockchain-build /out ${STACKS_BLOCKCHAIN_DIR}
+COPY --from=stacks-blockchain-api-build /app ${STACKS_BLOCKCHAIN_API_DIR}
+RUN cp ${STACKS_BLOCKCHAIN_API_DIR}/stacks-blockchain/Stacks-mocknet.toml ${STACKS_BLOCKCHAIN_DIR}/Stacks-testnet.toml \
+    && cp ${STACKS_BLOCKCHAIN_API_DIR}/stacks-blockchain/Stacks-mocknet.toml  ${STACKS_BLOCKCHAIN_DIR}/Stacks-mocknet.toml 
+
+###################################
+##  runit service files
+RUN printf '#!/bin/sh\nexec 2>&1\n[ ! -d %s ] && mkdir -p %s && chown -R postgres:postgres %s && gosu postgres /usr/lib/postgresql/%s/bin/pg_ctl init -D %s\nexec gosu postgres /usr/lib/postgresql/%s/bin/postmaster -D %s' ${PG_DATA} ${PG_DATA} ${PG_DATA} ${PG_VERSION} ${PG_DATA} ${PG_VERSION} ${PG_DATA} > ${STACKS_SVC_DIR}/postgresql/run \
```

Comment:
Have you considered committing the ruinit service files to this repo, and `COPY`ing them into the image? Might be easier to maintain and read that way.

Reply:
Possible, sure - but changes to the repo would require a full rebuild since the cache layers would no longer be used. 
Since there are many parts of this process, the goal was to create an image that can be built using cache layers as quickly as is feasible.

Reply:
> but changes to the repo would require a full rebuild since the cache layers would no longer be used.

If you `COPY` only those ruinit files, changes to the repo wouldn't require a full rebuild. That only happens when you write `COPY . .` for example. Either way will work the same, it's just difficult to read or maintain the scripts when they're embedded.